src/System.Management.Automation/utils/StringUtil.cs

```diff
@@ -81,6 +81,24 @@ internal static string Padding(int countOfSpaces)
 
             return result;
         }
+
+        private const int DashCacheMax = 120;
+        private static readonly string[] DashCache = new string[DashCacheMax];
+        internal static string DashPadding(int count)
+        {
+            if (count >= DashCacheMax)
+                return new string('-', count);
+
+            var result = DashCache[count];
+
+            if (result == null)
+            {
+                Interlocked.CompareExchange(ref DashCache[count], new string('-', count), null);
+                result = DashCache[count];
+            }
```

Comment:
How about change this if block to the following? You can return the new array.
```
if (result == null)
{
    result = new string('-', count);
    Interlocked.CompareExchange(ref DashCache[count], result, null);
}
```

Reply:
From a performance point of view, it shouldn't matter - the code should be hit just once per process in the normal case.

That said, returning the object from the cache is usually more desirable, and as a code pattern, I think that is reason enough to prefer the original code.

Reply:
> the code should be hit just once per process in the normal case.

This if block may be hit 120 times per process, since the array has 120 slots.
I'm fine with the original code. This is not a blocking comment.